src/transport/Session.cpp

```diff
@@ -71,9 +71,10 @@ System::Clock::Timeout Session::ComputeRoundTripTimeout(System::Clock::Timeout u
         return System::Clock::kZero;
     }
 
-    // Treat us as active for purposes of GetMessageReceiptTimeout(), since the
-    // other side would be responding to our message.
-    return GetAckTimeout() + upperlayerProcessingTimeout + GetMessageReceiptTimeout(System::SystemClock().GetMonotonicTimestamp());
+    // Treat us as active for purposes of GetMessageReceiptTimeout(), pass false into GetAckTimeout to
+    // indicate we are processing non-initial message since the other side would be responding to our message.
```

Comment:
No, the "responding to our message" is the `GetMessageReceiptTimeout()` bit, and it passes false already.  But GetAckTimeout.... we don't know what to pass it, because it depends on whether this ComputeRoundTripTimeout() call is for an initial message or not.  As in, this function needs a isFirstMessageOnExchange argument, just like the copy of this function in CASESession.cpp needed one.

Reply:
audited and update all ComputeRoundTripTimeout caller, thanks